### PR TITLE
feat: add observe plan information support

### PR DESCRIPTION
**Pull request by qingzhuozhen**
### Summary

add observe support 

### Checklist

* [ ] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-iOS/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:

### CODE REVIEW

**Review by justin-fiedler — APPROVED** [review]

File: Sources/Amplitude/Amplitude.m [hunk 1]
```diff
@@ -658,6 +660,10 @@ - (void)annotateEvent:(NSMutableDictionary *)event {
     [event setValue:library forKey:@"library"];
     [event setValue:[AMPUtils generateUUID] forKey:@"uuid"];
     [event setValue:[NSNumber numberWithLongLong:[self getNextSequenceNumber]] forKey:@"sequence_number"];
+
```
Comment (yuhao900914) [inline Sources/Amplitude/Amplitude.m:663]: Since we have a new key `plan` in the event payload. Dose this change also depends on our backend team?

Comment (qingzhuozhen) [reply Sources/Amplitude/Amplitude.m:663]: @yuhao900914 yeah, I guess Jeffery will handle the backend change

**Review by yuhao900914 — APPROVED** [review]

**Comment by haoliu-amp** [comment]
Would it be easy to fix the unit tests failures?

**Comment by yuhao900914** [comment]
> Would it be easy to fix the unit tests failures? If it takes too much efforts at this moment, forget it and create a separate ticket for it.

Yea, that will be great. Actually, it works fine on my local, but the test failed to build in GitHub action. Still trying to figure out what's the issue.

**Comment by qingzhuozhen** [comment]
Closing this, release is complete now, going to resolve conflict and merge to main

**Pull request closed without merging**